polygon.go

```diff
@@ -48,6 +53,13 @@ func (p *Polygon) Contains(point *Point) bool {
 		return false
 	}
 
+	// For now it recognizes a point only if it is in the polygon or on an edge. If the point is on one of the corners it returns false, so a quick and dirty array comparison.
```

Comment:
gotcha.  Is there a more efficient algorithm, method, or technique we could use here such that we're covering this use case, but not enumerating over the entire set of polygon points?

We don't have to implement it now, but would love to know a preferred path forward for future approaches.  Maybe a comment or something?

Reply:
I think a linear search should be fine as we are just searching the array once. (Nearly - there might be exceptions) All other search algorithms I know need a sorted array.

From [this discussion on stackoverflow](https://stackoverflow.com/questions/7656791/fastest-way-to-search-for-an-element-in-unsorted-array):
> 
> You can search an element with O(1) using this approach.
> 
> Just create a MAP . When you insert a value just then for that key assign value to '1', and to search it again just check if that array is present or not .
> https://stackoverflow.com/a/52225931/11708088


> 
> If you're searching for one element once, just iterate through it. No possible way to get it faster.
> 
> If you're searching multiple times, it would be worth it to index it (or sort it, if you will) and make the following searches fast (log(n)).
> https://stackoverflow.com/a/7656814/11708088



> Make it parallel. Divide the array in chunks and search in parallel. The complexity will be O(n) but running time will be much less. Actually it will be proportional to no. of processors you have.
> 
> You can use Parallel Patterns Library in C++
> https://stackoverflow.com/a/7656825/11708088

So the following methods could make a performance difference:

1. make it parallel
2. sorting and some search algorithm (just used once, so it takes too much time)
3. mapping

1. and 3. would need a benchmark test/comparison which could be done with *testing.B. Maybe a seperate test file with different implementations would work here.

Reply:
```suggestion
	// For now it recognizes a point only if it is in the polygon or on an edge. If the point is on one of the corners it returns false, so a quick and dirty array comparison.
	// Look here for further options: https://github.com/kellydunn/golang-geo/pull/71#discussion_r303040014
```